src/dftbp/timedep/timeprop.F90

```diff
@@ -1695,13 +1836,26 @@ subroutine getTDFunction(this, startTime)
       end if
 
       if (this%tEnvFromFile) then
+      #:if WITH_MPI
+        if (env%mpi%tGlobalLead) then
+          read(laserDat%unit, *)time, tdfun(1), tdfun(2), tdfun(3)
+        end if
+      #:else
         read(laserDat%unit, *)time, tdfun(1), tdfun(2), tdfun(3)
+      #:endif
         this%tdFunction(:, iStep) = tdfun * (Bohr__AA / Hartree__eV)
```

Comment:
Note: `tdfun(:)` has not been filled up on the follow ranks! Is that OK? (If yes, we should add at least a comment about this...)

Reply:
Good catch Bálint, and thanks for the review. We'll work on the changes ASAP so we can move on.